packages/luna/src/components/Chat/SendInput/SendInput.tsx

```diff
@@ -2,17 +2,23 @@ import React, { ChangeEvent, useCallback, useEffect, useState } from "react";
 import { useFormatMessage } from "@/hooks";
 import Send from "@litespace/assets/Send";
 import { Input } from "@/components/Input";
+import { Void } from "@litespace/types";
 
 type InitialMessage = { id: number; text: string };
 export const SendInput: React.FC<{
   initialMessage?: InitialMessage;
   onSubmit: (value: string) => void;
-}> = ({ initialMessage, onSubmit }) => {
+  typeMessage?: Void;
+}> = ({ initialMessage, onSubmit, typeMessage }) => {
   const [value, setValue] = useState<string>(initialMessage?.text || "");
   const intl = useFormatMessage();
-  const handleChange = useCallback((e: ChangeEvent<HTMLInputElement>) => {
-    setValue(e.target.value);
-  }, []);
+  const handleChange = useCallback(
+    (e: ChangeEvent<HTMLInputElement>) => {
+      setValue(e.target.value);
+      typeMessage ? typeMessage() : null;
```

Comment:
This is not clean. 

* This is not the right use case for the ternary operator.
* No need to `null`

```suggestion
    if (typeMessage) typeMessage();
```

Reply:
The problem with this approach is that we will send a typing event to the server each time the user **typed a new character**. Imaging user is typing an simple sentance that is just make of 50 char. This mean you unnecessarlly sent 50 event to the server just to reflect a simple state. You should down regular the number of events you send to the server to reflect this simple state. 


Here is how I will do it. I will keep an internal state of when I sent the last typing event. Each time user is typing an new character, I will check if an entire second has passed sense the typing event was sent.


Here is a sudo code for how to do this. 


```ts
const [typingNotice, setTypingNotice] = useState<Dayjs | null>(null);

const handleChange = useCallback((e) => {
    setValue(e.target.value);
    if (!typeMessage || (typingNotice && typingNotice.diff(dayjs(), 'ms') < 1000)) return 
    setTypingNotice(dayjs());
    typeMessage()
}, [])
```